pkg/server/plugin/nodeattestor/awsiid/eks.go

```diff
@@ -0,0 +1,315 @@
+package awsiid
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/andres-erbsen/clock"
+	"github.com/aws/aws-sdk-go-v2/service/autoscaling"
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/hashicorp/go-hclog"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type EKSClient interface {
+	eks.ListNodegroupsAPIClient
+	eks.DescribeNodegroupAPIClient
+}
+
+const (
+	eksNodeDefaultListTTL = "1m"  // pull account list after 1 minutes
+	eksNodeMinListTTL     = "10s" // Minimum TTL configuration to pull the org account list
+	eksRetries            = 5
+)
+
+var (
+	eksNodeDefaultListDuration, _ = time.ParseDuration(eksNodeDefaultListTTL)
+	eksNodeMinTTL, _              = time.ParseDuration(eksNodeMinListTTL)
+)
+
+type eksValidationConfig struct {
+	EKSClusterName string `hcl:"eks_cluster_name"`
+
+	NodesListTTL string `hcl:"eks_nodes_map_ttl"`
```

Comment:
I would prefer this not be a user-controlled configuration if we can avoid it. Can we find a reasonable default value?

Reply:
Will remove it from the config file, the idea came from keeping it similar to the account check